### PR TITLE
Doc updates for mc RELEASE.2024-01-05T05-04-32Z

### PR DESCRIPTION
**Pull request by feorlen**
Doc updates for these `mc` changes:

- New option for `mc mirror`: `--summary`
- Clarify `mc sql --json-input` also works with `.ndjson`

NDJSON clarified with a discussion in `mc sql` and an example of setting `content-type` in `mc cp`. 

Staged:
http://192.241.195.202:9000/staging/DOCS-1104/linux/reference/minio-mc/mc-mirror.html#mc.mirror.-summary

http://192.241.195.202:9000/staging/DOCS-1104/linux/reference/minio-mc/mc-sql.html#mc.sql.-json-input
http://192.241.195.202:9000/staging/DOCS-1104/linux/reference/minio-mc/mc-sql.html#input-formats
http://192.241.195.202:9000/staging/DOCS-1104/linux/reference/minio-mc/mc-cp.html#add-a-content-type-value

Fixes https://github.com/minio/docs/issues/1104

### CODE REVIEW

File: source/reference/minio-mc/mc-sql.rst [hunk 1]
```diff
@@ -145,7 +145,7 @@ Parameters
 .. mc-cmd:: --json-input
    
 
-   *Optional* The data format for ``.json`` input objects. Specify the type of 
+   *Optional* The data format for ``.json`` or ``.ndjson`` input objects. Specify the type of
```
Comment (djwfyi) [inline source/reference/minio-mc/mc-sql.rst:148]: Should we use the `:optional:` syntax here?
You fixed the other file, but may not want to fix all of them in this file. Up to you.

**Review by djwfyi — APPROVED** [review]
Consider fixing the optional/required thing in `mc sql` too.
But that's optional.

LGTM otherwise.

File: source/reference/minio-mc/mc-sql.rst (hunk 1 above)
Comment (feorlen) [reply source/reference/minio-mc/mc-sql.rst:148]: I contemplated but ended up not, as this file had _something_ at least.

It's not difficult, yeah might as well. I'll do it before merge, holding this overnight to see if there's any final eng input. 👍🏻

File: source/reference/minio-mc/mc-cp.rst [hunk 2]
```diff
@@ -449,6 +449,38 @@ command only applies to S3-to-S3 copy.
    :start-after: start-versioning-admonition
    :end-before: end-versioning-admonition
 
+Add a ``content-type`` Value
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use :mc-cmd:`mc cp --attr` to add a ``content-type`` value.
```
Comment (feorlen) [inline source/reference/minio-mc/mc-cp.rst:455]: Unclear if this should also have `This command only applies to S3-to-S3 copy` as some of the other examples do. Are there restrictions where you can do this?

**Comment by feorlen** [comment]
Slack fail, trying it this way

I clarified using content-type for objects with supported content but unexpected names. (.ndjson vs .json.) Mainly I'd appreciate a look from @jiuker. But there's a fair bit of new text, if you have a moment @ravindk89 and @djwfyi as well.

File: source/reference/minio-mc/mc-cp.rst (hunk 2 above)
Comment (ravindk89) [reply source/reference/minio-mc/mc-cp.rst:455]: :thinking: probably. I doubt we persist metadata to disk when copying from S3 to POSIX.

File: source/reference/minio-mc/mc-sql.rst [hunk 3]
```diff
@@ -228,28 +208,29 @@ Input Formats
...
 
-For ``.json`` file types, use :mc-cmd:`mc sql --json-input` to specify
-the JSON data format.
+:mc:`mc sql` determines the type by the file extension of the target object.
+For example, an object named ``data.json`` is interpreted as a JSON file.
 
-For ``.parquet`` file types, :mc:`mc sql` automatically interprets the
-data format.
+You can query data of a supported type but a different extension if the object has the appropriate ``content-type``.
+For more information, see :mc-cmd:`mc cp --attr`.
```
Comment (ravindk89) [inline source/reference/minio-mc/mc-sql.rst:222]: I wonder if we should include the explicit full attribute for content-type.

That is, the table on 207 could be a list table specifying the input format or the equivalent content type for the file.

**Review by ravindk89 — APPROVED** [review]
Still LGTM, but left some nits.

File: source/reference/minio-mc/mc-sql.rst (hunk 3 above)
Comment (feorlen) [reply source/reference/minio-mc/mc-sql.rst:222]: That would be handy, yes. So:
- `text/csv`
-  `application/json`
-  Whatever Parquet is.

Hmm [Parquet not yet](https://issues.apache.org/jira/browse/PARQUET-1889?focusedCommentId=17698080&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17698080), it seems

Comment (ravindk89) [reply source/reference/minio-mc/mc-sql.rst:222]: Hah. I guess we can leave that cell blank for now then.

**Review by jiuker — APPROVED** [review]

File: source/reference/minio-mc/mc-sql.rst [hunk 4]
```diff
@@ -106,71 +102,62 @@ Parameters
...
+   - For ``KEY``, specify the MinIO deployment :mc-cmd:`alias <mc alias>` and full path to the bucket, including any bucket prefixes.
+     Separate the alias and bucket path with a forward slash ``\``.
+     For example, ``play/mybucket``.
 
-   :mc-cmd:`~mc sql --encrypt-key` can use the ``MC_ENCRYPT_KEY``
-   environment variable for populating the list of encryption key-value
-   pairs as an alternative to specifying them on the command line.
+   - For ``VALUE``, specify the data key to use for encryption object(s) in the bucket or bucket prefix specified to ``KEY``.
+
+   :mc-cmd:`~mc sql --encrypt-key` can use the ``MC_ENCRYPT_KEY`` environment variable for populating the list of encryption key-value pairs as an alternative to specifying them on the command line.
```
Comment (djwfyi) [inline source/reference/minio-mc/mc-sql.rst:133]:
```suggestion
   :mc-cmd:`~mc sql --encrypt-key` can use the :envvar:`MC_ENCRYPT_KEY` environment variable for populating the list of encryption key-value pairs as an alternative to specifying them on the command line.
```

**Review by djwfyi — APPROVED** [review]
One suggestion to link to the environment variable, otherwise LGTM.

File: source/reference/minio-mc/mc-sql.rst (hunk 4 above)
Comment (feorlen) [reply source/reference/minio-mc/mc-sql.rst:133]: Ooo there's one in `mc cp` also. Might as well. 👍🏻

Comment (feorlen) [reply source/reference/minio-mc/mc-sql.rst]: Why why why is editing something through the GitHub UI so annoying? Fixing the things the normal way.

File: source/reference/minio-mc/mc-cp.rst [hunk 5]
```diff
@@ -58,7 +58,7 @@ similar results to the ``cp`` commandline tool.
                           [--attr "string"]                                         \
                           [--continue]                                              \
                           [--disable-multipart]                                     \
-                          [--encrypt "string"]                                      \
+                          [--encrytp "string"]                                      \
```
Comment (djwfyi) [inline source/reference/minio-mc/mc-cp.rst]: This was unintentional, I'd imagine.

```suggestion
                          [--encrypt "string"]                                      \
```

**Review by djwfyi — APPROVED** [review]

File: source/reference/minio-mc/mc-cp.rst (hunk 5 above)
Comment (feorlen) [reply source/reference/minio-mc/mc-cp.rst]: Apparently I used the emacs `mangle-words` command by accident. Fixed literally seconds before your comment.

**Pull request merged**